docs/credits.rst

```diff
@@ -227,7 +227,7 @@ Other Credits
 * Andrew Pontzen and the `pynbody <https://github.com/pynbody/pynbody>`_ team
   (For code that grew into :mod:`astropy.units`)
 * Everyone on `astropy-dev <http://groups.google.com/group/astropy-dev>`_
-  and the `astropy mailing list <http://mail.scipy.org/mailman/listinfo/astropy>`_
+  and the `astropy mailing list <https://mail.python.org/mailman/listinfo/astropy>`_
```

Comment:
@bsipocz - Good catch, thanks. The mailing lists links are defined in the `conf.py`, so I modified all references to use these instead.